static/js/click_handlers.js

```diff
@@ -11,7 +11,9 @@ exports.initialize = function () {
     function is_clickable_message_element(target) {
         return target.is("a") || target.is("img.message_inline_image") || target.is("img.twitter-avatar") ||
             target.is("div.message_length_controller") || target.is("textarea") || target.is("input") ||
-            target.is("i.edit_content_button") || target.is(".spoiler-arrow") ||
+            target.is("i.edit_content_button") ||
+            // For spoilers, allow clicking either the header or the dropdown arrow to expand it
+            target.is(".spoiler-header p") || target.is('.spoiler-arrow') ||
```

Comment:
I think the `p` bit here is going to be potentially problematic; it's possible to not have the outer tag be a `<p>` tag in Zulip markdown.  E.g. `~~~ spoiler * foo` will not have a `p` tag in the header.  I think it would be correct to include the entire `spoiler-header` here; we never will want to open the compose box in response to a click on the spoiler header, I think.

Reply:
Even with `p` I think it might not work if the header e.g. contains italic text and you click within the `<em>` tag, because then `em` is the target. I think we would need to check if `.spoiler-header` is a parent.

Reply:
The `p` tag when it exists obscures the `.spoiler-header`. This check is on the event target (i.e. the element actually clicked) so when there is a paragraph tag, the event is on it, not on `.spoiler-header`. We can probably solve this by checking if it is either `.spoiler-header` or a child of `.spoiler-header` though.

Reply:
Might be better to move this inside the `select_message_function` logic.

Reply:
@timabbott We changed the logic here and think it should handle the edge cases now.

Reply:
OK; I'll take a look at this some more.  We generally don't do any parent checks in that particular block; I'm not convinced that's actually important from a performance perspective.